src/geometryRTheta/time_solver/bsl_predcorr_second_order_implicit.hpp

```diff
@@ -297,7 +297,7 @@ class BslImplicitPredCorrRTheta : public ITimeSolverRTheta
 
 
             // STEP 4: From rho^P, we compute phi^P: Poisson equation
-            m_builder(get_field(allfdistribu_coef), get_const_field(allfdistribu_host));
+            m_builder(get_field(allfdistribu_coef), get_const_field(allfdistribu_predicted));
```

Comment:
There seems to be a problem here because `allfdistribu_predicted` is on the device while `allfdistribu_host` was on the host. Maybe a missing `ddc::create_mirror_view_and_copy`

```suggestion
            auto allfdistribu_predicted_host = ddc::create_mirror_view_and_copy(allfdistribu_predicted);
            m_builder(get_field(allfdistribu_coef), get_const_field(allfdistribu_predicted_host));
```